rdagent/oai/backend/litellm.py

```diff
@@ -212,3 +212,215 @@ def chat_token_limit(self) -> int:
             return max_tokens
         except Exception as e:
             return super().chat_token_limit
+
+    def supports_function_calling(self) -> bool:
+        """
+        Check if the backend supports function calling
+        """
+        return supports_function_calling(model=LITELLM_SETTINGS.chat_model)
+
+    def convert_mcp_tools_to_openai_format(self, mcp_tools: list) -> list[dict[str, Any]]:
+        """
+        Convert MCP tools to OpenAI function calling format
+
+        Args:
+            mcp_tools: List of MCP Tool objects
+
+        Returns:
+            List of tools in OpenAI format
+        """
+        openai_tools = []
+        for tool in mcp_tools:
+            openai_tool = {
+                "type": "function",
+                "function": {
+                    "name": tool.name,
+                    "description": tool.description,
+                    "parameters": tool.inputSchema,
+                },
+            }
+            openai_tools.append(openai_tool)
+
+        # Conversion completed silently
+        return openai_tools
+
+    def call_with_tools(
```

Comment:
suggest to make it a private method

Reply:
I think we can reuse  self._try_create_chat_completion_or_embedding. 
Otherwise, some features will be missing

Reply:
And the return value should be designed.
<img width="830" height="549" alt="image" src="https://github.com/user-attachments/assets/cdb1bdb3-16ed-4c5f-bdd7-0895b33235ae" />